server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -242,4 +244,36 @@ public void close() throws IOException {
             }
         }
     }
+
+    protected long getMinReferencedGen() throws IOException {
+        assert readLock.isHeldByCurrentThread() || writeLock.isHeldByCurrentThread();
+        long minReferencedGen = Math.min(
+            deletionPolicy.minTranslogGenRequired(readers, current),
+            minGenerationForSeqNo(Math.min(deletionPolicy.getLocalCheckpointOfSafeCommit() + 1, minSeqNoRequired), current, readers)
+        );
+        assert minReferencedGen >= getMinFileGeneration() : "deletion policy requires a minReferenceGen of ["
+            + minReferencedGen
+            + "] but the lowest gen available is ["
+            + getMinFileGeneration()
+            + "]";
+        assert minReferencedGen <= currentFileGeneration() : "deletion policy requires a minReferenceGen of ["
+            + minReferencedGen
+            + "] which is higher than the current generation ["
+            + currentFileGeneration()
+            + "]";
+        return minReferencedGen;
+    }
+
+    protected void minSeqNoRequired(long seqNo) {
+        this.minSeqNoRequired = seqNo;
+    }
+
+    void deleteReaderFiles(TranslogReader reader) {
+        try {
+            translogTransferManager.deleteTranslog(primaryTermSupplier.getAsLong(), reader.generation);
+        } catch (IOException ignored) {
+
+        }
+        super.deleteReaderFiles(reader);
```

Comment:
If delete in remote fails, do we still want to delete local files?

Reply:
Yes, we don't want to hold on to local files and filling the disk.  We can have a background thread however to clean up the crumbs left here. I will create an issue and update here.